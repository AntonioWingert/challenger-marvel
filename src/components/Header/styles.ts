import { styled } from "styled-components";

export const Container = styled.header`
  display: flex;
  justify-content: start;
  align-items: center;
  width: 100vw;
  height: 10vh;
  background-color: ${({theme}) => theme.colors['bg-primary']};
`;