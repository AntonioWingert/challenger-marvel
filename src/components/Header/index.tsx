import CustomTitle from "../CustomTitle";

import * as S from './styles';

export default function Header() {
  return (
    <S.Container>
      <CustomTitle title="Marvel Challenger" $size="lg" $weight="regular" $color="secondary"/>
    </S.Container>
  );
}