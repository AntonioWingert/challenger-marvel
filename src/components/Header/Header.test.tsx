import { screen } from '@testing-library/react';
import render from '../../utils/renderWithRouter';
import Header from '.';

describe('<Header />', () => {
  it('should render correctly with default props', () => {
    const { container } = render(
      <Header />
    );

    expect(screen.getByText(/Marvel Challenger/i)).toBeVisible();
    expect(screen.getByText(/Marvel Challenger/i)).toHaveStyleRule('font-size', '2.4rem');
    expect(container.firstChild).toMatchSnapshot();
  });
});