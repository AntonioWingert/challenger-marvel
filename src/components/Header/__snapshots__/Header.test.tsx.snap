// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > should render correctly with default props 1`] = `
.c1 {
  font-size: 2.4rem;
  color: #F5F5F5;
  font-weight: 400;
  padding: 32px;
}

.c0 {
  display: flex;
  justify-content: start;
  align-items: center;
  width: 100%;
  height: 10vh;
  background-color: #4A1414;
}

@media (max-width: 320px) {
  .c1 {
    font-size: 1.6rem;
  }
}

<header
  class="c0"
>
  <h1
    class="c1"
  >
    Marvel Challenger
  </h1>
</header>
`;
